ggshield/core/git_hooks/ci/get_scan_ci_parameters.py

```diff
@@ -0,0 +1,109 @@
+import os
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
+
+import click
+
+from ggshield.core.errors import NotAMergeRequestError, UnexpectedError
+from ggshield.utils.git_shell import get_commits_not_in_branch, get_remotes, git
+
+from .supported_ci import SupportedCI
+
+
+def travis_scan_ci_args() -> Tuple[str, str]:
+    if "TRAVIS_PULL_REQUEST" not in os.environ:
+        raise NotAMergeRequestError()
+    try:
+        commit_range = os.environ["TRAVIS_COMMIT_RANGE"]
+        first_commit, last_commit = commit_range.split("..")
+        return last_commit, first_commit + "~1"
+    except Exception as exc:
+        raise UnexpectedError("Failed to extract scan ci arguments for travis") from exc
+
+
+# Note: this does not exist (yet ?) for CircleCI, see
+# https://circleci.canny.io/config/p/provide-env-variable-for-branch-name-targeted-by-pull-request
+CI_TARGET_BRANCH_ASSOC: Dict[SupportedCI, str] = {
+    SupportedCI.GITHUB: "GITHUB_BASE_REF",
+    SupportedCI.GITLAB: "CI_MERGE_REQUEST_TARGET_BRANCH_NAME",
+    SupportedCI.JENKINS: "CHANGE_TARGET",
+    SupportedCI.AZURE: "SYSTEM_PULLREQUEST_TARGETBRANCHNAME",
+    SupportedCI.BITBUCKET: "BITBUCKET_PR_DESTINATION_BRANCH",
+    SupportedCI.DRONE: "DRONE_COMMIT_BRANCH",
+}
+
+
+def get_scan_ci_parameters(
+    current_ci: SupportedCI,
+    wd: Optional[Union[str, Path]] = None,
+    verbose: bool = False,
+) -> Union[Tuple[str, str], None]:
+    """
+    Function used to gather current commit and reference commit, for the SCA/IaC scan
+    ci commands.
+    Return: None if the MR has no associated commits, otherwise a tuple (current_commit, reference commit)
+    """
+    if verbose:
+        click.echo(
+            f"\tIdentified current ci as {current_ci.value}",
+            err=True,
+        )
+
+    if current_ci == SupportedCI.TRAVIS:
+        return travis_scan_ci_args()
+
+    # we will work with branch names, so we run a git fetch to ensure the names are locally available
+    git(["fetch"], cwd=wd)
```

Comment:
I understand it's handy to have the fetch here, but I think it would be better to have the caller do this, for two reasons:

- The function name does not look like a function which would alter the repository it is working on, this could lead to surprising behavior. Especially since right now it's not done for Travis.
- If we add more functions which require an up-to-date repository, and they each call `git fetch`, then we end up with multiple fetches where we only need one.

Reply:
It would really worsen the UX though - WDYT of fetching at the command level ? (so that we only do it once)

Also the cost of subsequent fetch is not negligible, but still much less than the first one.

Reply:
Fetching at the command level sounds good to me :+1:.

Reply:
Thanks, i  cleanup up the commits, should be ready then :)